web/modules/custom/dpl_library_agency/src/Form/GeneralSettingsForm.php

```diff
@@ -144,6 +144,27 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
       '#default_value' => $config->get('pause_reservation_start_date_config'),
     ];
 
+    $form['settings']['blocked_user'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Blocked user'),
+      '#collapsible' => FALSE,
+      '#collapsed' => FALSE,
+    ];
+
+    $form['settings']['blocked_user']['redirect_on_blocked_url'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Redirect blocked user link'),
+      '#description' => $this->t('The link to redirect the blocked user to'),
+      '#default_value' => $config->get('redirect_on_blocked_url') ?? '',
+    ];
+
+    $form['settings']['blocked_user']['blocked_patron_e_link_url'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Blocked user link for modal'),
+      '#description' => $this->t('If a user has blocked status e, this link appears in the modal'),
+      '#default_value' => $config->get('blocked_patron_e_link_url') ?? '',
+    ];
```

Comment:
I think you should use the Form API type: 'url' here:
https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21Element%21Url.php/class/Url/

Reply:
Then you get the url validation for free! :)

Reply:
Changed